### ISSUE TITLE
Applying Html.App.map twice leads to "TypeError: domNode.replaceData is not a function"

### ISSUE THREAD

**Issue by marmutro** [elm-lang/elm-compiler#1382]
Hi,

I managed to reproduce a bug by modifiying the button example :

It tags the messages of a button with Html.App.map into another type, and then tags (wraps) this value again with Html.App.map

```
import Html exposing (div, button, text)
import Html.App exposing (beginnerProgram)
import Html.Events exposing (onClick)

type alias Model =
    Int

type ButtonMsg
    = Increment
    | Decrement

type ItemMsg
    = Button ButtonMsg

type Msg
    = Item ItemMsg

main =
    beginnerProgram { model = 0, view = view, update = update }

buttonView : Model -> Html.Html ButtonMsg
buttonView model =
    div []
        [ button [ onClick Decrement ] [ text "-" ]
        , div [] [ text (toString model) ]
        , button [ onClick Increment ] [ text "+" ]
        ]

view : Model -> Html.Html Msg
view model =
    let
        itemView model =
            Html.App.map Button (buttonView model)
    in
        Html.App.map Item (itemView model)


update : Msg -> Model -> Model
update msg model =
    case msg of
        Item itemMsg ->
            case itemMsg of
                Button buttonmsg ->
                    case buttonmsg of
                        Increment ->
                            model + 1
                        Decrement ->
                            model - 1


```
Is this a bug of the compiler or the elm runtime?
see [elm-discuss](https://groups.google.com/forum/?fromgroups#!topic/elm-discuss/qTJQrMJqZbo)

Regards,
Martin

**Comment by jvoigtlaender** [comment]
Related discussion at https://github.com/elm-lang/html/issues/16.

**Comment by jvoigtlaender** [comment]
Can be closed here, since being worked on at https://github.com/elm-lang/virtual-dom/issues/21.

**Comment by evancz** [comment]
Should be fixed by `elm-lang/virtual-dom` 1.0.2
